app/controllers/lti_deployments_controller.rb

```diff
@@ -14,8 +14,9 @@ def launch
       return
     end
     nonce = rand(10 ** 30).to_s.rjust(30, '0')
-    session[:client_id] = params[:client_id]
-    session[:nonce] = nonce # Nonce should be present in the JWT sent to MarkUs, and must be verified
+    cookies.encrypted[:client_id] = { value: params[:client_id], expires: 1.hour.from_now }
```

Comment:
We talked about putting all info into a single cookie. Is that something that we still want to do?